dora_node_hub/dora_string_publisher_ui/src/main.rs

```diff
@@ -0,0 +1,21 @@
+use dora_node_api::DoraNode;
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let (mut node, mut _events) = DoraNode::init_from_env()?;
+    let options = eframe::NativeOptions { ..Default::default() };
+
+    eframe::run_native(
+        "Dora Text Publisher",
+        options,
+        Box::new(|cc| {
```

Comment:
@francocipollone meta: so a box with a closure that returns a box. Is it because it needs box dyn polymorphism? I'm just curious.

Reply:
That parameter expects an `AppCreator` that is defined as follows
```rust
pub type AppCreator<'app> =
    Box<dyn 'app + FnOnce(&CreationContext<'_>) -> Result<Box<dyn 'app + App>, DynError>>;
```
 - See that you need a box of a type that implements [FnOnce](https://doc.rust-lang.org/std/ops/trait.FnOnce.html) trait. 
 - And you need to return a box of a type that implements `eframe::App` trait.

So you kind of create a functor of your own application that is heap allocated.
That's at least at an API level, not sure under the hood why they used this pattern.